### ISSUE TITLE
Warp caused - "The specified key does not exist"

### ISSUE THREAD

**Issue by akmithal** [noobaa/noobaa-core#6733]
### Environment info
- NooBaa Version: master-20210927
- Platform: OCP 4.8.9
### Actual behavior
1. IO failure errors

### Expected behavior
1. No IO failure should come 

### Steps to reproduce
1. Warp caused IO failure:

```
[root@hpo-11 all-sizes]# warp mixed --access-key=BrVzxXwyhi470cqA3s0c --secret-key=h+9u0jszQGENpsstCar3XQ7nYlFHUgzHj03V01Wr --obj.size=9M --host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com --duration=2h --bucket=bucket-1 --debug --tls --insecure
warp: <ERROR> {  The specified key does not exist.}
warp: <ERROR> {  EOF}
warp: <ERROR> {  The specified key does not exist.}
warp: <ERROR> {  EOF}
warp: <ERROR> The specified key does not exist.
```

```
Sep-27 13:12:57.627 [Endpoint/13] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Resource>/bucket-1/?delete=</Resource><RequestId>ku2o872l-agiwql-wxl</RequestId></Error> POST /bucket-1/?delete= {"user-agent":"MinIO (linux; amd64) minio-go/v7.0.11 warp/0.3.43","content-length":"60346","authorization":"AWS4-HMAC-SHA256 Credential=BrVzxXwyhi470cqA3s0c/20210927/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=4f1f5902acb94b1d334441de05455e9a55d822e51ed1a5457cabb72b7ef705ab","content-md5":"Op5Rg6dpPiLX3SE+H/5OgA==","x-amz-content-sha256":"4e533133f75547f29c720d098c4e000a42e02a8f316562249851c09a819f5d9e","x-amz-date":"20210927T131257Z","host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-port":"443","x-forwarded-proto":"https","forwarded":"for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https","x-forwarded-for":"10.17.66.193"} Error: No such file or directory
2021-09-27 13:12:57.629487 [PID-13/TID-13] [L1] FS::FSWorker::OnError: Rmdir _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ  error.Message()=Directory not empty 
Sep-27 13:12:57.629 [Endpoint/13]    [L0] core.sdk.namespace_fs:: NamespaceFS: delete_multiple_objects /nsfs/nsfs-nsr-1/bucket-1/F61mljDJ/72.T0benbBP5A98ysdn.rnd
2021-09-27 13:12:57.629701 [PID-13/TID-13] [L1] FS::FSWorker::Begin: Unlink _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ/72.T0benbBP5A98ysdn.rnd 
2021-09-27 13:12:57.629740 [PID-13/TID-21] [L1] FS::FSWorker::Execute: Unlink _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ/72.T0benbBP5A98ysdn.rnd _uid=5000 _gid=800 _backend=GPFS 
2021-09-27 13:12:57.631097 [PID-13/TID-21] [L1] FS::FSWorker::Execute: Unlink _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ/72.T0benbBP5A98ysdn.rnd  took: 1.32184 ms
2021-09-27 13:12:57.636111 [PID-13/TID-13] [L1] FS::FSWorker::Begin: FileWritev _wrap->_path=/nsfs/nsfs-nsr-1/bucket-3/.noobaa-nsfs_6151bae35fd0290029f0086c/multipart-uploads/df0362a4-6cf6-463d-8e1d-c4f8f51d9970/part-1 total_len=8388608 buffers_len=156 
2021-09-27 13:12:57.636170 [PID-13/TID-21] [L1] FS::FSWorker::Execute: FileWritev _wrap->_path=/nsfs/nsfs-nsr-1/bucket-3/.noobaa-nsfs_6151bae35fd0290029f0086c/multipart-uploads/df0362a4-6cf6-463d-8e1d-c4f8f51d9970/part-1 total_len=8388608 buffers_len=156 _uid=5002 _gid=800 _backend=GPFS 
2021-09-27 13:12:57.637662 [PID-13/TID-13] [L1] FS::FSWorker::OnOK: undefined Unlink _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ/72.T0benbBP5A98ysdn.rnd 
2021-09-27 13:12:57.637725 [PID-13/TID-13] [L1] FS::FSWorker::Begin: Rmdir _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ 
2021-09-27 13:12:57.638558 [PID-13/TID-23] [L1] FS::FSWorker::Execute: Rmdir _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ _uid=5000 _gid=800 _backend=GPFS 
2021-09-27 13:12:57.638818 [PID-13/TID-23] [L1] FS::FSWorker::Execute: Rmdir _path=/nsfs/nsfs-nsr-1/bucket-1/F61mljDJ  took: 0.185933 ms
2021-09-27 13:12:57.642480 [PID-13/TID-21] [L1] FS::FSWorker::Execute: FileWritev _wrap->_path=/nsfs/nsfs-nsr-1/bucket-3/.noobaa-nsfs_6151bae35fd0290029f0086c/multipart-uploads/df0362a4-6cf6-463d-8e1d-c4f8f51d9970/part-1 total_len=8388608 buffers_len=156  took: 6.28318 ms

```

**Comment by akmithal** [comment]
[must-gather-2709_2.tar.gz](https://github.com/noobaa/noobaa-core/files/7236531/must-gather-2709_2.tar.gz)

**Comment by nimrod-becker** [comment]
@akmithal how does this differs from #6704 ?

**Comment by akmithal** [comment]
This bug comes very infrequently, so do you think basic triaging should happen on either of them to find the root cause. 
We can track it through a single defect.

**Comment by akmithal** [comment]
This defect is not a duplicate of 6704 and is getting fixed as part of 6756.

**Comment by akmithal** [comment]
@nimrod-becker : Please add label for Request validation here.

**Comment by akmithal** [comment]
This issue seems to be fixed with ODF 4.9.4-7 build. Closing this defect.
